### ISSUE TITLE
Repository URL fails to parse when CodeArtifact domain contains a '-'

### ISSUE THREAD

**Issue by randalpitt** [dangernoodle-io/codeartifact-maven-wagon#9]
Great extension! We'd love to make use of this, but unfortunately we get an error when Maven attempts to access the repository.

Our CodeArtifact domain has the structure `foo-bar` and the URL, as returned by `aws codeartifact get-repository-endpoint`, has the structure `https://foo-bar-000123456789.d.codeartifact...`. 

Defining a repository with that URL then using this extension results in the following error:

```
[ERROR] 2 validation errors detected: Value 'bar' at 'domainOwner' failed to satisfy constraint: Member must 
have length greater than or equal to 12; Value 'bar' at 'domainOwner' failed to satisfy constraint: Member 
must satisfy regular expression pattern: [0-9]{12} (Service: Codeartifact, Status Code: 400, Request ID: ...)
```

I believe it is splitting the URL on the `-` and assuming the second element is always the domain owner.

**Comment by chrisxaustin** [comment]
[CodeArtifact](https://github.com/dangernoodle-io/codeartifact-maven-extension/blob/c79b773be9966958af2e8e50589f7f3cc9d5058c/src/main/java/io/dangernoodle/codeartifact/maven/CodeArtifact.java#L63) has
```
private GetAuthorizationTokenRequest createTokenRequest(String domainOwner)
    {
        String[] parts = domainOwner.split("-");
        return GetAuthorizationTokenRequest.builder()
                                           .domain(parts[0])
                                           .domainOwner(parts[1])
                                           .durationSeconds(getTokenDuration())
                                           .build();
    }
```

According to [this](https://docs.aws.amazon.com/codeartifact/latest/ug/domain-overview.html) the domain owner is only needed when you're not authenticated to the account that owns the domain

**Comment by tonistaht** [comment]
I have opened a PR to fix this issue 
https://github.com/dangernoodle-io/codeartifact-maven-extension/pull/11

**Comment by jgangemi** [comment]
hello - i apologize for such a delayed response! 

i am in the process of getting a few things sorted regarding maven central and the release process. once we that's done i will merge your pull-request and publish a new version.

again, i apologize for this taking so long and appreciate the use of the plugin!!

**Comment by jgangemi** [comment]
resolved via #15
